### ISSUE TITLE
The role of proxy (in future)

### ISSUE THREAD

**Issue by wenbozhu** [MikeBishop/http-layering#2]
Does it make sense to limit the role of proxies to just the transport layer?

This also makes it easier to decompose HTTP into the two layers: end-to-end semantic layer v.s. hop-by-hop transport layer.

But again, transport != L4 under the OSI model.

**Comment by MikeBishop** [comment]
I don't think so, though I haven't attempted to discuss proxies in the draft so far.  There's certainly a lot that could be said.  Of course, it depends what type of proxy we're talking about.  Reverse proxies are the endpoint as far as the client is concerned -- how it satisfies the request via backend services is irrelevant to the client.  Then there are transport-layer proxies -- but unless they're changing transports, that may (should) be transparent to HTTP.

Forward proxies are an explicit concept that we should think about and discuss, I suspect.  They actually are part of Semantic HTTP, even if you don't like them.  Using this issue to track adding a proxy conversation to the semantic section.

**Comment by wenbozhu** [comment]
Right. HTTP proxy likely has to support/understand both layers.

Reverse proxies, if acting as an transport proxies (quite likely), would only need understand the transport layer (if such a concept got defined in future). 

Agreed that forward proxies needs understand the HTTP semantics.
